server/src/controllers/ticketTypeController.ts

```diff
@@ -242,33 +288,64 @@ ticketTypeController.get('/:id', async (req: Request, res: Response) => {
 ticketTypeController.put('/:id', async (req: Request, res: Response) => {
   try {
     const id = req.params.id;
-    const ticketType = prisma.tickettype.update({
+    const {description, price, concessions} = req.body;
+
+    if (price === undefined || +price < 0 || concessions === undefined || +concessions < 0) {
+      return res.status(400).json({error: `Neither price nor concession price can be negative`});
+    } else if (!description|| description === '') {
+      return res.status(400).json({error: `Ticket type description is required`});
+    }
+
+    const ticketType = await prisma.tickettype.findUnique({
       where: {
-        tickettypeid: Number(id),
+        tickettypeid: +id,
+      },
+    });
+
+    if (!ticketType) {
+      return res.status(400).json({error: `Ticket type does not exist`});
+    }
+
+    const updatedType = await prisma.tickettype.update({
+      where: {
+        tickettypeid: +id,
       },
       data: {
-        description: req.body.description,
-        price: req.body.price,
-        concessions: req.body.concessions,
-        deprecated: req.body.deprecated,
+        description,
+        price: !id? 0: +price,
+        concessions: +concessions,
+        ticketrestrictions: {
+          updateMany: {
+            where: {
+              tickettypeid_fk: +id,
+            },
+            data: {
+              ...(ticketType.tickettypeid && +ticketType.price !== +price && {price: +price}),
+              ...(+ticketType.concessions !== +concessions && {concessionprice: +concessions}),
```

Comment:
I don't think it would actually matter if we check whether or not the price/concession prices are the same?

Reply:
I think if we don't check every time an update (like a name change) is made it will automatically cascade the existing price down to the ticket restrictions/season ticket prices.

Reply:
Oh, ok that’s good then